manifests.go

```diff
@@ -0,0 +1,49 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// Manifest specifies a registry object with a list of dependencies.
+type Manifest interface {
+	Blob // io access to serialized content.
+
+	// Dependencies returns a list of object on which this manifest depends.
+	// The dependencies are strictly ordered from base to head. Typically,
+	// these are layers but their interpretation is application specific.
+	Dependencies() []Descriptor
```

Comment:
@icecrime After some thought, I think the answer is both. A raw blob service shouldn't have to know about its contents. This should be something we can support. However, if a use case requires traversal, one should be able to wrap the blob service in a component that detect's media types and returns interesting objects, such as a manifest, that could be switched upon.